storm/src/main/storm/mesos/schedulers/DefaultScheduler.java

```diff
@@ -54,6 +55,52 @@ public void prepare(Map conf) {
     mesosStormConf = conf;
   }
 
+  private List<MesosWorkerSlot> getMesosWorkerSlots(Map<String, OfferResources> offerResourcesPerNode,
+                                                    Collection<String> nodesWithExistingSupervisors,
+                                                    TopologyDetails topologyDetails) {
+
+    double requestedWorkerCpu = MesosCommon.topologyWorkerCpu(mesosStormConf, topologyDetails);
+    double requestedWorkerMem = MesosCommon.topologyWorkerMem(mesosStormConf, topologyDetails);
+
+    List<MesosWorkerSlot> mesosWorkerSlots = new ArrayList<>();
+    boolean slotFound = true;
+    int slotsNeeded = topologyDetails.getNumWorkers();
+
+    while (slotFound && slotsNeeded > 0) {
```

Comment:
If you can propose a different formulation that isn't *wrong* then I'd be ok with not using do-while, I suppose you have similar feels about it that I do about `unless` in ruby.  But the previous style where we set slotFound to true when we haven't found one yet is just really misleading.